src/Interpreters/IInterpreterUnionOrSelectQuery.h

```diff
@@ -10,11 +10,12 @@ namespace DB
 class IInterpreterUnionOrSelectQuery : public IInterpreter
 {
 public:
-    IInterpreterUnionOrSelectQuery(const ASTPtr & query_ptr_, ContextPtr context_, const SelectQueryOptions & options_)
+    IInterpreterUnionOrSelectQuery(const ASTPtr & query_ptr_, ContextPtr context_, const SelectQueryOptions & options_, bool is_explain_ = false)
```

Comment:
Here, `is_explain` is also not a very good name.
Also, it is logical to put it into `SelectQueryOptions`.
Also, `SelectQueryOptions` already has setting `only_analyze`. Maybe we should better set it for explain (and probably do not create sets when enabled).